AK/FixedPoint.h

```diff
@@ -202,27 +204,38 @@ class FixedPoint {
     }
     constexpr This operator*(This const& other) const
     {
-        // FIXME: Potential Overflow, although result could be represented accurately
-        Underlying value = m_value * other.raw();
-        This ret {};
-        ret.raw() = value >> precision;
-        // fract(value) >= .5?
+        // FIXME: Figure out a way to use more narrow types and avoid __int128
+        using MulRes = Conditional<sizeof(Underlying) < sizeof(i64), i64, __int128>;
+
+        MulRes value = raw();
+        value *= other.raw();
```

Comment:
```suggestion
        auto value = UFixedBigInt<Detail::bit_width<Underlying>>(raw()).wide_multiply(other.raw());
```

Reply:
Oops, `Underlying` can be signed... Then it will, unfortunately, require manual handling of the signs.

Reply:
Yeah I did not find an algorithm for signed wide multiply, otherwise I'd have used that and made that a generic helper

Reply:
Actually it looks as if we must use unsigned types here. The next operation is the shift, so it is UB if the result turns out to be negative.
`baseline_mul` from `BigIntBase.h` theoretically can do signed multiplies but that is not tested at all and the API is not particularly easy to use.